tools/commands-packages-query.js

```diff
@@ -1053,22 +1053,43 @@ main.registerCommand({
     new catalog.Refresh.OnceAtStart(
         { maxAge: DEFAULT_MAX_AGE_MS, ignoreErrors: true })
 }, function (options) {
-  // The foo@bar API means that we have to do some string parsing to figure out
-  // if we want a particular version.
-  var itemName = options.args[0];
-  var splitArgs = itemName.split('@');
-  var name = splitArgs[0];
-  var version = (splitArgs.length > 1) ? splitArgs[1] : null;
-  if (splitArgs.length > 2) {
-    Console.error("Invalid request format: " + itemName);
-    process.exit(1);
-  }
-  var query = null;
-
+  var fullName;
+  var name;
+  var version;
   // Because of the new projectContext interface, we need to initialize the
   // project context in order to load the local catalog. This is not ideal.
   var projectContext = getTempContext(options);
 
+  // If the user specified a query, process it.
+  if (! _.isEmpty(options.args)) {
+    // The foo@bar API means that we have to do some string parsing to figure out
+    // if we want a particular version.
+    fullName = options.args[0];
+    var splitArgs = fullName.split('@');
+    name = splitArgs[0];
+    version = (splitArgs.length > 1) ? splitArgs[1] : null;
+    if (splitArgs.length > 2) {
+      Console.error("Invalid request format: " + fullName);
+      process.exit(1);
+    }
+  } else {
+    if (! options.packageDir) {
+      // Letting the user run 'meteor show' without arguments from a package
+      // directory is a pleasant shortcut, but the default should be specifying
+      // a query.
+      Console.error(
+        "Please specify a package or release name to show information about it."
+      );
+      process.exit(1);
+    }
+    var currentVersion =
+          projectContext.localCatalog.getVersionBySourceRoot(options.packageDir);
+    name = currentVersion.packageName;
+    version = currentVersion.version;
+    fullName = name + "@" + version;
```

Comment:
Yeah, "@local" would also work here. There is no real difference.

I think my original motivation was wanting to be really careful to make it clear to people that there is a local version, and that's what they will end up using. To be honest, the semantics were worked out before 'meteor show foo@local' became a thing.

Here is what I am thinking: there is a workflow where you say "Wow, I can't add foo:baz because of foo:bar? Meteor list! I am using foo:bar at 2.0.0? What is in foo:bar at 2.0.0? Wait, it is *local*? Huh." 
- Understanding that you should run 'meteor show foo:bar@local' from 'meteor list' is unintuitive. We could provide a message at the bottom to try and limit the confusion, I guess.
- It is weird to show you information about a version that you can't use. At least, it seems like a strange default. It would be nice to have a better override.
- How often do you really need to look at a server version of a package version that you are already mirroring locally? (Especially since you probably have the git repository handy, since you forked it to begin with.) I am actually not very sure -- you might want to check out the newer version, but the same version? Is this use case common enough to outweigh the other concerns?

I also think that it is reasonable, longterm, to consider implementing some sort of a --remote option, or better yet IGNORE_LOCAL_PKGS environment variable (for the super advanced users). That would always ignore local packages, which you might want if you are ... running some sort of a package reference script? Meteor from checkout and don't remember the state of the published versions? It seems worthwhile to wait until there are requests for that.

I might be totally off-base though. Do you think the 'no specification' makes it more confusing than it was before? I can take it out.